### DISCUSSION TITLE
Add support for laptops

### DISCUSSION THREAD

**Discussion by pankaps** [Rem0o/FanControl.Releases#2533] [Ideas]
I have an  Acer Aspire 7 A715-75g and software can't recognize fans,i installed HWlnfo plugin, i get message  "HWlnfo could not initialize or has no sensors.
NVApi could not initialize or has no sensors.".
Then i saw it doesn't support laptop.Will we ever get support for laptops?

**Comment by Rem0o** [comment]
> Will we ever get support for laptops?

Probably not.  Laptops are not meant for manual fan control. Manufacturer do not design them with that in mind.  All laptops have a specific fan system, almost all of them have severe control limitation.  Each type of laptop is a project in and of itself, and finding ways/code to control fans on laptops is a treasure hunt. 

FanControl have a plugin system, so technically, anyone can contribute and add support for anything, that's not the hard part.  Interfacing FanControl is easy.  
 
I did develop support for "some" dell laptops with this https://github.com/Rem0o/FanControl.DellPlugin, however, it's backed by a library which is a reverse engineered hack and has serious limitations.  For eg, on my laptop, the system will eventually downclock itself if I set manual control from it, even if the temps are within their normal operating range. Also, there are only 3 possible speeds: off, low and high, so forget about fancy tuning and graphs.  That's for (some) dell laptops. Every other manufacturer is a different complicated story. 

Hope you get the idea.

**Reply by pankaps** [reply]
Ok, thanks for response.How hard is it to make my own plugin?Any alternative i could use for laptops?
